tips/TIP-0025/core-rest-api.yaml

```diff
@@ -0,0 +1,2701 @@
+      example:
+        error:
+          code: 403
+          message: not available for public use
+
+    ServiceUnavailableResponse:
+      description: Indicates that the service is unavailable.
+      allOf:
+        - $ref: '#/components/schemas/ErrorResponse'
+      example:
+        error:
+          code: 503
+          message: service unavailable
+
+    BadRequestResponse:
+      description: Indicates that the request was bad.
+      allOf:
+        - $ref: '#/components/schemas/ErrorResponse'
+      example:
+        error:
+          code: 400
+          message: invalid data provided
+
+    NotFoundResponse:
+      description: Indicates that the data was not found.
+      allOf:
+        - $ref: '#/components/schemas/ErrorResponse'
+      example:
+        error:
+          code: 404
+          message: could not find data
+
+    InternalErrorResponse:
+      description: Indicates that the server encountered an unexpected condition, which prevented it from fulfilling the request by the client.
+      allOf:
+        - $ref: '#/components/schemas/ErrorResponse'
+      example:
+        error:
+          code: 500
+          message: internal server error
+
+    InfoResponse:
+      description: Returns general information about the node.
+      properties:
+        name:
+          type: string
+          description: The name of the node.
+        version:
+          type: string
+          description: The semantic version of the node.
+        status:
+          description: Status of the node.
+          properties:
+            isHealthy:
+              type: boolean
+              description: Tells whether the node is healthy or not.
+            latestMilestoneTimestamp:
+              type: integer
+              description: The timestamp of the latest seen milestone.
+            latestMilestoneIndex:
+              type: integer
+              description: The most recent milestone known to the node.
+            confirmedMilestoneIndex:
+              type: integer
+              description: The most recent milestone that has been confirmed by the node.
+            pruningIndex:
+              type: integer
+              description: Tells from which starting point the node holds data.
+        metrics:
+          description: Node metrics.
+          properties:
+            messagesPerSecond:
+              description: The current rate of new messages per second.
+              type: number
+              format: float
+            referencedMessagesPerSecond:
+              description: The current rate of referenced messages per second.
+              type: number
+              format: float
+            referencedRate:
+              description: The ratio of referenced messages in relation to new messages of the last confirmed milestone.
+              type: number
+              format: float
+        protocol:
+          description: Protocol parameters.
+          properties:
```

Comment:
We should extend this model with:
- `version` reflecting the protocol version
- `tokenSupply` reflecting the hex encoded token supply

as these are dynamic parameters of the protocol.

Reply:
Never mind, you're right, the `tokenSupply` is supposed to be a plain string, not hex encoded.